### PR TITLE
Installer revendor to cdb9627 (4.11)

### PR DESCRIPTION
**Pull request by abutcher**
Update our installer module dependency to `cdb9627`:
- Adds a new Azure client function `GetVMCapabilities` which returns a `capabilities` `map[string]string` of capabilities an instanceType supports in a region. Installer machineset generation for Azure now accepts this capabilities map as a parameter.
- Updates the default Azure worker instanceType to `Standard_D4s_v3` to match the installer default. Capabilities for the previous instanceType of `Standard_D2s_v3` in the `centralus` region report accelerated networking as available but creating an instance of type `Standard_D2s_v3` with accelerated networking results in the following error creating the machine:
  ```
  failed to reconcile machine "hive-093a30fc-42e1-45-bkhpz-infra-centralus3-8674w":
  failed to create nic hive-093a30fc-42e1-45-bkhpz-infra-centralus3-8674w-nic for machine hive-093a30fc-42e1-45-bkhpz-infra-centralus3-8674w:
  accelerated networking not supported on instance type: Standard_D2s_v3'
  ```

[HIVE-1972](https://issues.redhat.com//browse/HIVE-1972)
/hold

### CODE REVIEW

File: pkg/azureclient/client.go [hunk 1]
```diff
@@ -131,29 +130,22 @@ func (c *azureClient) StartVirtualMachine(ctx context.Context, resourceGroup, na
 	return c.virtualMachinesClient.Start(ctx, resourceGroup, name)
 }
 
-// GetHyperVGenerationVersion gets the HyperVGeneration version for the given disk instance type. Defaults to V2 if either V1 or V2
-// available.
-func (c *azureClient) GetHyperVGenerationVersion(ctx context.Context, instanceType string, diskType string, region string) (version string, err error) {
+// GetVMCapabilities retrieves the capabilities of an instance type in a specific region. Returns these values
```
Comment (2uasimojo) [inline pkg/azureclient/client.go:133]: copyin from [installer](https://github.com/openshift/hive/blob/81367fc2cad747d3101d950ce80d2b5fa634046d/vendor/github.com/openshift/installer/pkg/asset/installconfig/azure/client.go#L279-L295) ✓

File: pkg/controller/machinepool/azureactuator.go [hunk 2]
```diff
@@ -101,17 +105,14 @@ func (a *AzureActuator) GenerateMachineSets(cd *hivev1.ClusterDeployment, pool *
...
-				cd.Spec.Platform.Azure.Region,
-			)
-			if err != nil {
-				return nil, false, err
+		if !gen2ImageExists {
+			// Modify capabilities to ensure that a V1 image is chosen by installazure.MachineSets
+			// because a gen2 image does not exist.
+			// The HyperVGeneration is germane to the instance/disk type and affects the image used
+			// for the instance. "-gen-2" will be appended to the image name when HyperVGenerations
+			// capability includes "V2".
```
Comment (2uasimojo) [inline pkg/controller/machinepool/azureactuator.go:113]: I think this piece of commentary was left over from a previous iteration where we were string scraping to figure out the version?

File: pkg/controller/machinepool/azureactuator_test.go [hunk 3]
```diff
@@ -153,8 +160,9 @@ func TestAzureActuator(t *testing.T) {
 			pool:              testAzurePool(),
 			mockAzureClient: func(mockCtrl *gomock.Controller, client *mockazure.MockClient) {
 				mockListResourceSKUs(mockCtrl, client, []string{"zone1", "zone2", "zone3"})
-				mockGetHyperVGeneration(mockCtrl, client, "V2")
+				mockGetVMCapabilities(mockCtrl, client, "V1,V2")
```
Comment (2uasimojo) [inline pkg/controller/machinepool/azureactuator_test.go:163]: Should there be a test case where only V1 exists? Similarly for only V2?

**Review by 2uasimojo — COMMENTED** [review]
looking good.

File: pkg/controller/machinepool/azureactuator.go (hunk 2 above)
Comment (abutcher) [reply pkg/controller/machinepool/azureactuator.go:113]: The `installer.MachineSets()` call below will append `"-gen2"` to the image name based on `"HyperVGenerations"`  within the `capabilities` map. I'll make that more clear in the comment.

File: pkg/controller/machinepool/azureactuator_test.go (hunk 3 above)
Comment (abutcher) [reply pkg/controller/machinepool/azureactuator_test.go:186]: Added tests for only V1 or only V2. As discussed offline, I think we need a FUP around this that future proofs the implementation with the expectation that V3 images will be introduced in the future.

**Comment by abutcher** [comment]
Monitoring e2e-azure.

**Comment by abutcher** [comment]
Digging into this.

```
    machineSets:
    - errorMessage: 'failed to reconcile machine "hive-093a30fc-42e1-45-bkhpz-infra-centralus3-8674w":
        failed to create nic hive-093a30fc-42e1-45-bkhpz-infra-centralus3-8674w-nic
        for machine hive-093a30fc-42e1-45-bkhpz-infra-centralus3-8674w: accelerated
        networking not supported on instance type: Standard_D2s_v3'
      errorReason: InvalidConfiguration
      maxReplicas: 1
      minReplicas: 1
      name: hive-093a30fc-42e1-45-bkhpz-infra-centralus3
      replicas: 1
```

**Comment by abutcher** [comment]
The above e2e failure can be addressed by bumping the default instance type to `Standard_D4s_v3` to match the installer's default. I've tested creating infra machinepools with this new default on 4.8, 4.9 and 4.10 Azure spokes.

**Pull request merged**